### PR TITLE
Prep for 1.0; bump go to 1.10.3 (new build image); linter fix; refactor official build; misc

### LINKED ISSUE #252: move release build details from Dockerfile to Makefile
so they can be used more easily for other binary release like brew

https://github.com/Homebrew/homebrew-core/pull/26653#issuecomment-391075797

### PR DESCRIPTION
**Pull request by ldemailly**
Prep for 1.0; also bump go to 1.10.3 / new build image and fixes
previous pr broke the build update and missed 1 file
and gometalinter doesn't report errors when using go list canonical package names (istio.io/fortio/x)

Also moved logic for official builds from Dockerfile to Makefile, fixes #252 

And better docs #234

### CODE REVIEW

**Comment by ldemailly** [comment]
there are more doc changes and build change I want to do but I need to get the linters fix in asap, ptal

File: fgrpc/grpcrunner.go [hunk 1]
```diff
@@ -50,7 +50,8 @@ func Dial(serverAddr, cacert, override string) (conn *grpc.ClientConn, err error
 	var opts []grpc.DialOption
 	switch {
 	case cacert != "":
-		creds, err := credentials.NewClientTLSFromFile(cacert, override)
+		var creds credentials.TransportCredentials
+		creds, err = credentials.NewClientTLSFromFile(cacert, override)
```
Comment (ymesika) [inline fgrpc/grpcrunner.go:54]: Is this change a lint requirement now?

File: ui/uihandler.go [hunk 2]
```diff
@@ -962,37 +962,28 @@ func Report(baseurl, port, staticRsrcDir string, datadir string) bool {
...
-		portStr = fmt.Sprintf(":%d", addr.Port)
-	}
-	if !strings.Contains(inputPort, ":") {
-		inputPort = ":" + inputPort
-	}
-	if strings.HasPrefix(inputPort, ":") {
-		urlHostPort = "localhost" + portStr
+	urlHostPort = addr.String()
+	if strings.HasPrefix(inputPort, ":") || !strings.Contains(inputPort, ":") {
+		urlHostPort = fmt.Sprintf("localhost:%d", addr.Port)
```
Comment (ymesika) [inline ui/uihandler.go:971]: Just to clarify - you expect `add.Port` to be the same as the provided `inputPort` (numerically)?

**Review by ymesika — APPROVED** [review]

File: fgrpc/grpcrunner.go (hunk 1 above)
Comment (ldemailly) [reply fgrpc/grpcrunner.go:54]: yes err was shadowed

File: ui/uihandler.go (hunk 2 above)
Comment (ldemailly) [reply ui/uihandler.go:971]: inputPort could be a service name (e.g http or smtp etc...) that is used to bind and resolved internally by go but generally yes

**Comment by ldemailly** [comment]
screen shot update:
screen shot 2018-06-14 at 11 22 25 am

**Comment by ldemailly** [comment]
and

screen shot 2018-06-14 at 11 25 41 am

**Pull request merged**